File: src/python/PythonSDK/foundationallm/hubs/prompt/prompt_hub_request.py
from typing import Optional
from pydantic import BaseModel

class PromptHubRequest(BaseModel):
    """     
    PromptHubRequest contains the information needed to retrieve prompts from the PromptHub.
    
    prompt_container: The prompt container from which prompt values will be retrieved.
                Send an empty string or None to retrieve all prompts.
    session_id: the session ID value; can be used for caching.
    """
    prompt_container: Optional[str] = None
    session_id:Optional[str] = None
    prompt_name: Optional[str] = 'default'

File: src/dotnet/Common/Models/Hubs/PromptHubRequest.cs
using System.Text.Json.Serialization;

namespace FoundationaLLM.Common.Models.Hubs
{
    /// <summary>
    /// The format of a Prompt Hub request.
    /// </summary>
    public record PromptHubRequest
    {
        /// <summary>
        /// The session ID.
        /// </summary>
        [JsonPropertyName("session_id")]
        public string? SessionId { get; set; }

        /// <summary>
        /// The prompt container from which prompt values will be retrieved.
        /// </summary>
        [JsonPropertyName("prompt_container")]
        public string? PromptContainer { get; set; }

        /// <summary>
        /// Name of the prompt for which the prompt values should be retrieved from the Prompt Hub.
        /// </summary>
        [JsonPropertyName("prompt_name")]
        public string? PromptName { get; set; } = "default";
    }
}
